src/tasks/DockerBuildTaskProvider.ts

```diff
@@ -0,0 +1,115 @@
+                },
+                task));
+    }
+
+    // TODO: Skip if image is freshly built
+    private async resolveTaskInternal(context: DockerBuildTaskContext, task: DockerBuildTask): Promise<Task> {
+        context.actionContext.telemetry.properties.platform = context.platform;
+
+        const definition = cloneObject(task.definition);
+        definition.dockerBuild = definition.dockerBuild || {};
+
+        if (!context.folder) {
+            throw new Error(`Unable to determine task scope to execute docker-build task '${task.name}'.`);
+        }
+
+        const helper = this.getHelper(context.platform);
+
+        definition.dockerBuild = await helper.resolveDockerBuildOptions(context, definition);
+        await this.validateResolvedDefinition(context, definition.dockerBuild);
+
+        const commandLine = await this.resolveCommandLine(definition.dockerBuild);
+        // TODO : addDockerSettingsToEnv?
+        return new Task(
+            task.definition,
+            task.scope,
+            task.name,
+            task.source,
+            new ShellExecution(commandLine[0], commandLine.slice(1)),
+            task.problemMatchers);
+    }
+
+    private async validateResolvedDefinition(context: DockerBuildTaskContext, dockerBuild: DockerBuildOptions): Promise<void> {
+        if (!dockerBuild.tag) {
+            throw new Error('No Docker image name was resolved.');
+        }
+
+        if (!dockerBuild.context) {
+            throw new Error('No Docker build context was resolved.');
+        } else if (!await fse.pathExists(resolveWorkspaceFolderPath(context.folder, dockerBuild.context))) {
+            throw new Error(`The Docker build context \'${dockerBuild.context}\' does not exist or could not be accessed.`);
+        }
+
+        if (!dockerBuild.dockerfile) {
+            throw new Error('No Dockerfile was resolved.');
+        } else if (!await fse.pathExists(resolveWorkspaceFolderPath(context.folder, dockerBuild.dockerfile))) {
+            throw new Error(`The Dockerfile \'${dockerBuild.dockerfile}\' does not exist or could not be accessed.`);
+        }
+    }
+
+    private async resolveCommandLine(options: DockerBuildOptions): Promise<ShellQuotedString[]> {
+        return CommandLineBuilder
+            .create('docker', 'build', '--rm')
+            .withFlagArg('--pull', options.pull)
+            .withNamedArg('-f', options.dockerfile)
+            .withKeyValueArgs('--build-arg', options.args)
+            .withKeyValueArgs('--label', options.labels)
+            .withNamedArg('-t', options.tag)
+            .withNamedArg('--target', options.target)
+            .withQuotedArg(options.context)
+            .buildShellQuotedStrings();
+    }
+
+    private getHelper(platform: DockerPlatform): TaskHelper {
+        const helper = this.helpers[platform];
+
+        if (!helper) {
+            throw new Error(`The platform '${platform}' is not currently supported for Docker build tasks.`);

```

Comment:
@philliphoff It occurs to me that if we allow for no helper, they can use our tasks to define any imaginable docker-build / docker-run. As long as they provide sufficient information to build/start the container, we don't technically need to infer anything extra.

So, I think we should _not_ fail if no helper is available. (For tasks that is, but not for debug)

Reply:
The benefit isn't huge; they can already do this via any arbitrary commandline task, but it lets them take advantage of our intellisense on task definition to more easily create a functional command line.